test/e2e/framework/framework.go

```diff
@@ -273,3 +263,17 @@ func createNamespace(client kubeclientset.Interface, name string, labels map[str
 	Expect(err).NotTo(HaveOccurred(), "Error creating namespace %v", namespaceObj)
 	return namespace
 }
+
+func (f *Framework) GenerateNamespace(name string, namespacelables map[string]string) string {
+	for idx, clientSet := range f.ClusterClients {
+		switch idx {
+		case ClusterA: // On the first cluster we let k8s generate a name for the namespace
+			namespace := generateNamespace(clientSet, f.BaseName, namespacelables)
+			f.Namespace = namespace.GetName()
```

Comment:
This assignment should live on the other function, so Subsequent generated namespaces won't modify the main framework "Namespace" or the "UniqueName"

Reply:
ok then I assume `namespace` should be declared before line 268 and passed on line 275 below?  I guess the assumption is that `ClusterA` is the first element in `f.ClusterClients`.

Reply:
Yeah, that's an option, would be cleaner. Although it's not part of janki's patch itself.